### PR TITLE
Add the "NOT IN" operator to `queryForListFilter()`.

### PR DESCRIPTION
**Pull request by ordavidil**
#1051

### CODE REVIEW

**Comment by ordavidil** [comment]
@amitaibu - ready for review

File: tests/RestfulListTestCase.test [hunk 1]
```diff
@@ -524,6 +524,34 @@ class RestfulListTestCase extends RestfulCurlBaseTestCase {
     $result = $handler->get('', $request);
     $this->assertEqual(count($result), 3, 'List filtered by the "author" entity metadata wrapper property, which maps to the "uid" DB column name, using the "IN" operator.');
 
+    // Filter by an entity metadata wrapper property that is different from the
+    // DB column name using the 'IN' operator.
```
Comment (amitaibu) [inline tests/RestfulListTestCase.test]: IN => NOT IN

File: tests/RestfulListTestCase.test [hunk 2]
```diff
@@ -524,6 +524,34 @@ class RestfulListTestCase extends RestfulCurlBaseTestCase {
...
+          'operator' => 'NOT IN',
+        ),
+      ),
+    );
+
+    $result = $handler->get('', $request);
+    $this->assertEqual(count($result), 0, 'List filtered by the "author" entity metadata wrapper property, which maps to the "uid" DB column name, using the "NOT IN" operator.');
+
+    // Filter by an entity metadata wrapper property that is different from the
+    // DB column name using the 'IN' operator.
```
Comment (amitaibu) [inline tests/RestfulListTestCase.test]: IN => NOT IN

**Review by amitaibu — COMMENTED** [review]
Looks good, thanks! Minor typos, and it should be ready to go.

**Comment by ordavidil** [comment]
@amitaibu - ready for final review

**Comment by amitaibu** [comment]
Thanks.

**Comment by amitaibu** [comment]
I'll create a new release later

**Pull request merged**